sycl/plugins/level_zero/CMakeLists.txt

```diff
@@ -1,143 +1,102 @@
-    INSTALL_DIR       "${CMAKE_CURRENT_BINARY_DIR}/level_zero_loader_install"
-    CMAKE_ARGS -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}
-               -DCMAKE_MAKE_PROGRAM=${CMAKE_MAKE_PROGRAM}
-               -DCMAKE_INSTALL_PREFIX=<INSTALL_DIR>
-               -DCMAKE_INSTALL_LIBDIR:PATH=lib${LLVM_LIBDIR_SUFFIX}
-               ${AUX_CMAKE_FLAGS}
-               LOG_DOWNLOAD 1
-               LOG_UPDATE 1
-               LOG_CONFIGURE 1
-               LOG_BUILD 1
-               LOG_INSTALL 1
-    STEP_TARGETS      configure,build,install
-    DEPENDS           OpenCL-Headers
-    BUILD_BYPRODUCTS ${LEVEL_ZERO_LOADER}
-  )
-  ExternalProject_Add_Step(level-zero-loader llvminstall
-    COMMAND ${CMAKE_COMMAND} -E copy_directory <INSTALL_DIR>/lib/ ${LLVM_BINARY_DIR}/lib
-    COMMAND ${CMAKE_COMMAND} -E copy_directory <INSTALL_DIR>/include/ ${LLVM_BINARY_DIR}/include/sycl
-    COMMENT "Installing level-zero-loader into the LLVM binary directory"
-    DEPENDEES install
-  )
 
-  install(DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/level_zero_loader_install/"
-    DESTINATION "."
-    COMPONENT level-zero-loader
-  )
+  set(LEVEL_ZERO_LOADER_REPO "https://github.com/oneapi-src/level-zero.git")
+  set(LEVEL_ZERO_LOADER_TAG v1.8.8)
 
-  set(LEVEL_ZERO_INCLUDE_DIR ${CMAKE_CURRENT_BINARY_DIR}/level_zero_loader_install/include/
-	  CACHE INTERNAL "Path containing Level_Zero header files.")
-else()
-  file(GLOB LEVEL_ZERO_LIBRARY_SRC "${LEVEL_ZERO_LIBRARY}*")
-  get_filename_component(LEVEL_ZERO_LIB_NAME ${LEVEL_ZERO_LIBRARY} NAME)
-  add_custom_target(level-zero-loader
-      DEPENDS
-	  ${LLVM_LIBRARY_OUTPUT_INTDIR}/${LEVEL_ZERO_LIB_NAME}
-	  ${LLVM_BINARY_DIR}/include/sycl/level_zero
-  )
-  add_custom_command(
-      OUTPUT
-	  ${LLVM_LIBRARY_OUTPUT_INTDIR}/${LEVEL_ZERO_LIB_NAME}
-	  ${LLVM_BINARY_DIR}/include/sycl/level_zero
-      COMMENT
-	  "Copying Level Zero loader and headers"
-      COMMAND
-	  ${CMAKE_COMMAND} -E copy ${LEVEL_ZERO_LIBRARY_SRC} ${LLVM_LIBRARY_OUTPUT_INTDIR}
-      COMMAND
-	  ${CMAKE_COMMAND} -E copy_directory ${LEVEL_ZERO_INCLUDE_DIR} ${LLVM_BINARY_DIR}/include/sycl
-      DEPENDS
-	  ${LEVEL_ZERO_LIBRARY}
-          ${LEVEL_ZERO_INCLUDE_DIR}
+  # Disable due to a bug https://github.com/oneapi-src/level-zero/issues/104
+  set(CMAKE_INCLUDE_CURRENT_DIR OFF)
+
+  message(STATUS "Will fetch Level Zero Loader from ${LEVEL_ZERO_LOADER_REPO}")
+  include(FetchContent)
+  FetchContent_Declare(level-zero-loader
+    GIT_REPOSITORY    ${LEVEL_ZERO_LOADER_REPO}
+    GIT_TAG           ${LEVEL_ZERO_LOADER_TAG}
   )
-endif()
 
-list(APPEND SYCL_TOOLCHAIN_DEPLOY_COMPONENTS level-zero-loader level-zero-headers)
+  # Suppress some pedantic warnings for Level Zero build
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-unused-but-set-variable")
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-pedantic")
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-stringop-truncation")
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-unknown-warning-option")
```

Comment:
Can you try target_compile_definitions(ze_loader ... ) instead?
I assume we want these flags for external level_zero project, rather than globally for our project.

Reply:
The target_compile_definitions & target_compile_options say that it "cannot specify compile options for target which is not built by this project"

I tried to restore original flags in https://github.com/intel/llvm/pull/7521/commits/08efaed555fa45f381763c56bb85f90744c62929

Reply:
You need to add it after ze_loader is defined. See 7031ac02b0290b1778ea90ed208c43e8284ec5d4

Reply:
while it seems to be working you'll have to list all internal level_zero targets :(
So solution you had is probably better.